care/utils/assetintegration/onvif.py

```diff
@@ -39,19 +45,89 @@ def handle_action(self, action):
+            return self.request_access(username, asset_id)
+
+        if action_type == BaseAssetIntegration.BaseAssetActions.UNLOCK_ASSET.value:
+            if self.unlock_camera(username, asset_id):
+                return {"message": "Asset Unlocked"}
+            self.raise_conflict(asset_id=asset_id)
+
+        if action_type == BaseAssetIntegration.BaseAssetActions.LOCK_ASSET.value:
+            if self.lock_camera(username, asset_id):
+                return {"message": "Asset Locked"}
+            self.raise_conflict(asset_id=asset_id)
+
         if action_type == self.OnvifActions.GET_CAMERA_STATUS.value:
+            self.lock_camera(username, asset_id)
             return self.api_get(self.get_url("status"), request_body)
 
         if action_type == self.OnvifActions.GET_PRESETS.value:
+            self.lock_camera(username, asset_id)
             return self.api_get(self.get_url("presets"), request_body)
 
         if action_type == self.OnvifActions.GOTO_PRESET.value:
-            return self.api_post(self.get_url("gotoPreset"), request_body)
+            if self.verify_access(username, asset_id):
+                self.lock_camera(username, asset_id)
+                return self.api_post(self.get_url("gotoPreset"), request_body)
+            self.raise_conflict(asset_id=asset_id)
 
         if action_type == self.OnvifActions.ABSOLUTE_MOVE.value:
-            return self.api_post(self.get_url("absoluteMove"), request_body)
+            if self.verify_access(username, asset_id):
+                self.lock_camera(username, asset_id)
+                return self.api_post(self.get_url("absoluteMove"), request_body)
+            self.raise_conflict(asset_id=asset_id)
 
         if action_type == self.OnvifActions.RELATIVE_MOVE.value:
-            return self.api_post(self.get_url("relativeMove"), request_body)
+            if self.verify_access(username, asset_id):
+                self.lock_camera(username, asset_id)
+                return self.api_post(self.get_url("relativeMove"), request_body)
+            self.raise_conflict(asset_id=asset_id)
 
         raise ValidationError({"action": "invalid action type"})
+
+    def validate_action(self, action):
+        from care.facility.models.bed import AssetBed, ConsultationBed
+
+        action_type = action["type"]
+        action_data = action.get("data", {})
+        boundary_preset_id = action_data.get("id", None)
+
+        if (
+            not boundary_preset_id
+            or action_type != self.OnvifActions.RELATIVE_MOVE.value
+        ):
+            return
+
+        boundary_preset = AssetBed.objects.get(external_id=boundary_preset_id)
+
+        # check if privacy mode is on
+        consultation_bed = ConsultationBed.objects.filter(
+            bed=boundary_preset.bed, privacy=True
+        ).first()
+
+        if (
+            consultation_bed
+            and consultation_bed.privacy
+            and consultation_bed.meta["locked_by"] != action["user"]
```

Comment:
```suggestion
            and consultation_bed.meta.get("locked_by") != action["user"]
```
handle key error on locked_by